src/colony/modules/_net_proxied.js

```diff
@@ -0,0 +1,200 @@
+      "MIICazCCAdQCCQDr2mJoysZo9DANBgkqhkiG9w0BAQUFADB6MQswCQYDVQQGEwJV",
+      "UzELMAkGA1UECBMCQ0ExETAPBgNVBAcTCEJlcmtlbGV5MQ8wDQYDVQQKEwZUZXNz",
+      "ZWwxGDAWBgNVBAMTD3Byb3h5LnRlc3NlbC5pbzEgMB4GCSqGSIb3DQEJARYRdGVh",
+      "bUB0ZWNobmljYWwuaW8wHhcNMTUwMTIzMDExOTQ5WhcNMTUwMjIyMDExOTQ5WjB6",
+      "MQswCQYDVQQGEwJVUzELMAkGA1UECBMCQ0ExETAPBgNVBAcTCEJlcmtlbGV5MQ8w",
+      "DQYDVQQKEwZUZXNzZWwxGDAWBgNVBAMTD3Byb3h5LnRlc3NlbC5pbzEgMB4GCSqG",
+      "SIb3DQEJARYRdGVhbUB0ZWNobmljYWwuaW8wgZ8wDQYJKoZIhvcNAQEBBQADgY0A",
+      "MIGJAoGBAMBZDd+DvwOxtCHl0tVYjctfoim7GvVrE257vofr7oi1SXGXf1ZPcPgb",
+      "DDiq5zq5G38JaHNHpkUq5+J8XUNXgdITAm0Pj4RHwwIHejmWh7ZWRZqsxrz+E6T0",
+      "aj+RIqe3wAfNmr4N7pZ8un9XL1abcTREOMdaLvsNBLhFSJOf3B2tAgMBAAEwDQYJ",
+      "KoZIhvcNAQEFBQADgYEAbNbBFOe5P33FWUup35d/zUdU/q2j5xDGhMbyCVkdONh9",
+      "QrjD9jStLIhaLOjXleI5S+TT9Vgqknn7D9Q977+/wTAg6aXwSBe1LK/V4du92eSx",
+      "/stxtxpFH7BQ1YPuuzmIcYd8rGPywjV91fsgqmWIZFe7FHGbRqKNKPCeR3Nv4tg=",
+      "-----END CERTIFICATE-----",
+    ].join('\n');
+
+/**
+ * Tunnel helpers
+ */
+
+function createTunnel(cb) {
+  if (_PROXY_DBG) console.log("TUNNEL -> START", new Date());
+  tls.connect({host:PROXY_HOST, port:PROXY_PORT, proxy:false, ca:[PROXY_CERT]}, function () {
+    var proxySocket = this,
+        tunnel = streamplex(streamplex.B_SIDE);
+    tunnel.pipe(proxySocket).pipe(tunnel);
+    proxySocket.on('error', shutdownTunnel);
+    proxySocket.on('close', shutdownTunnel);
+    proxySocket.on('error', cb);
+    
+    var idleTimeout;
+    tunnel.on('inactive', function () {
+      if (_PROXY_DBG) console.log("TUNNEL -> inactive", new Date());
+      idleTimeout = setTimeout(shutdownTunnel, PROXY_IDLE);
+    });
+    tunnel.on('active', function () {
+      if (_PROXY_DBG) console.log("TUNNEL -> active", new Date());
+      clearTimeout(idleTimeout);
+    });
+    
+    tunnel.sendMessage({token:PROXY_TOKEN});
+    tunnel.once('message', function (d) {
+      if (_PROXY_DBG) console.log("TUNNEL: auth response?", d);
+      proxySocket.removeListener('error', cb);
+      if (!d.authed) cb(new Error("Authorization failed."));
+      else cb(null, tunnel);
+    });
+    function shutdownTunnel(e) {
+      if (_PROXY_DBG) console.log("TUNNEL -> STOP", new Date());
+      tunnel.destroy(e);
+      if (this !== proxySocket) proxySocket.end();
+      proxySocket.removeListener('close', shutdownTunnel);
+    }
+  }).on('error', cb);
+}
+
+var tunnelKeeper = new events.EventEmitter();
+
+tunnelKeeper.getTunnel = function (cb) {    // CAUTION: syncronous callback!
+    if (this._tunnel) return cb(null, this._tunnel);
+    
+    var self = this;
+    if (!this._pending) createTunnel(function (e, tunnel) {
+      delete self._pending;
+      if (e) return self.emit('tunnel', e);
+      
+      self._tunnel = tunnel;
+      tunnel.on('close', function () {
+        self._tunnel = null;
+      });
+      var streamProto = Object.create(ProxiedSocket.prototype);
```

Comment:
`streamProto` becomes the prototype for an already-existing socket object (see @tcr's "I hate you" comment above ;-) and this bypasses calling the actual constructors again, only preparing the [think of it like a] "mix-in" we now need.

If it weren't for needing a link between socket instances and their associated tunnel (next two lines) the logic that uses this would just `ProxiedSocket.prototype` directly instead.